catshadow/client_docker_test.go

```diff
@@ -409,7 +409,10 @@ and can readily scale to millions of users.
 	// Test statefile persistence of conversation.
 
 	alice.log.Debug("LOADING ALICE'S CONVERSATION")
+	alice.conversationsMutex.Lock()
```

Comment:
@david415 wouldn't it be better to refactor this into a method , and since `map[MessageID]*Message` is also a map , shouldn't that also be an issue.
```go
func (c *Client) getConversation(nickname string) map[MessageID]*Message {
	c.conversationsMutex.Lock()
	defer c.conversationsMutex.Unlock()
	conv, ok := c.conversations[nickname]
	if !ok {
		return nil
	}

	conversationsCopy := make(map[MessageID]*Message)
	for k, v := range conv {
		conversationsCopy[k] = v
	}
	return conversationsCopy
}
```

Reply:
yes that's better